src/random.cpp

```diff
@@ -553,6 +577,16 @@ void GetRandBytes(unsigned char* buf, int num) noexcept { ProcRand(buf, num, RNG
 void GetStrongRandBytes(unsigned char* buf, int num) noexcept { ProcRand(buf, num, RNGLevel::SLOW); }
 void RandAddPeriodic() { ProcRand(nullptr, 0, RNGLevel::PERIODIC); }
 
+void RandAddEvent(const unsigned char* event_type_buf, size_t buf_len) {
+    LOCK(events_mutex);
+    events_hasher.Write((unsigned char *)&buf_len, sizeof(size_t));
```

Comment:
You could also limit this to 4 bytes (or even 2...), if size is an issue.

Reply:
Hmm, I'm not too worried about it? Its mostly up to the caller and right now net will only ever pass in 12+4+8 bytes per call, which means it will never be more than one extra hash per two messages, which seems reasonable.